packages/privacy-scan-core/src/cookie-collector.ts

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { injectable } from 'inversify';
+import _ from 'lodash';
+import { ConsentResult, CookieByDomain } from 'storage-documents';
+import * as Puppeteer from 'puppeteer';
+import { CookieScenario } from './cookie-scenarios';
+import { ReloadPageFunc, ReloadPageResponse } from '.';
+
+@injectable()
+export class CookieCollector {
+    public async getCookiesForScenario(
+        page: Puppeteer.Page,
+        cookieScenario: CookieScenario,
+        reloadPage: ReloadPageFunc,
```

Comment:
a bit confusing name convention as it is function, cam call similar to type `reloadPageFunc`